File: submissions/0128-longest-consecutive-sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        //392
        if(nums.length == 0) return 0;
        HashSet<Integer> set = new HashSet<>();
        int max = 1;
        int count = 1;
        for(int i=0; i<nums.length; i++){
            set.add(nums[i]);
        }
        for(int num : set){
            if(!set.contains(num-1)){
                int curr = num;
                count = 1;
                while(set.contains(curr + 1)){
                    //set.remove(curr);   i cannot remove and traverse elements from set at same time
                    count++;
                    curr++;
                }
                max = Math.max(max,count);
            }
        }
        //max = Math.max(max,count);
        return max;
    }
}

File: submissions/0128-longest-consecutive-sequence/solution.js
var longestConsecutive = function(nums) {
    const map = new Map();
    let longest = 0;

    for (let num of nums) {
        if (map.has(num)) continue;

        const left = map.get(num - 1) || 0;
        const right = map.get(num + 1) || 0;

        const total = left + right + 1;

        map.set(num, total);
        map.set(num - left, total);
        map.set(num + right, total);

        longest = Math.max(longest, total);
    }

    return longest;
};
